Neos.Fusion/Classes/FusionObjects/DictionaryImplementation.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace Neos\Fusion\FusionObjects;
+
+/*
+ * This file is part of the Neos.Fusion package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Fusion\Core\Runtime;
+
+/**
+ * Evaluate sub objects to an array (instead of a string as ConcatImplementation does)
+ */
+class DictionaryImplementation extends ArrayImplementation
```

Comment:
Shouldn't `DictionaryImplementation` extend `ConcatenationImplementation` directly, instead of the deprecated `ArrayImplementation`? But then again, a `Dictionary` really isn't a `Concatenation` in any way, so this hints at some improper naming or inheritance. So what's the common ground? The sorting. Actually, `Dictionary` doesn't do much else. What about `Dictionary` becoming a `Sorting` and `Concatenation` extending that?

Reply:
@albe the names try to describe the expected result instead of the operation or process. Imho this is what reflects the declarative nature of fusion best.